nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/SendTrapSNMP.java

```diff
@@ -0,0 +1,212 @@
+            V1TrapProperties.PREDEFINED_GENERIC_TRAP_TYPE,
+            V1TrapProperties.GENERIC_TRAP_TYPE,
+            V1TrapProperties.SPECIFIC_TRAP_TYPE,
+            V2TrapProperties.TRAP_OID_VALUE
+    ));
+
+    private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            REL_SUCCESS, REL_FAILURE
+    )));
+
+    private volatile SendTrapSNMPHandler snmpHandler;
+    private Instant startTime;
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext context) {
+        List<ValidationResult> results = new ArrayList<>();
+
+        final boolean isPredefinedGenTrapType = context.getProperty(V1TrapProperties.PREDEFINED_GENERIC_TRAP_TYPE).isSet();
+        final boolean isGenTrapType = context.getProperty(V1TrapProperties.GENERIC_TRAP_TYPE).isSet();
+        final boolean isV1Trap = SnmpConstants.version1 == context.getProperty(BasicProperties.SNMP_VERSION).asInteger();
+
+        if (isV1Trap && isPredefinedGenTrapType && isGenTrapType) {
+            results.add(new ValidationResult.Builder()
+                    .subject(V1TrapProperties.GENERIC_TRAP_TYPE.getDisplayName())
+                    .explanation("Only one Generic Trap Type property can be set.")
+                    .valid(false)
+                    .build());
+        } else if (isV1Trap && !isPredefinedGenTrapType && !isGenTrapType) {
+            results.add(new ValidationResult.Builder()
+                    .subject(V1TrapProperties.GENERIC_TRAP_TYPE.getDisplayName())
+                    .explanation("Generic Trap Type or Predefined Generic Trap Type must be set.")
+                    .valid(false)
+                    .build());
+        }
+
+        return results;
+    }
+
+    @OnScheduled
+    public void init(ProcessContext context) {
+        startTime = Instant.now();
+        initSnmpManager(context);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession processSession) {
+        final FlowFile flowFile = processSession.get();
+        final int snmpVersion = context.getProperty(BasicProperties.SNMP_VERSION).asInteger();
+        final Supplier<PDU> pduFactory;
+        if (SnmpConstants.version1 == snmpVersion) {
+            V1TrapConfiguration trapConfiguration = V1TrapConfiguration.builder()
+                    .enterpriseOid(context.getProperty(V1TrapProperties.ENTERPRISE_OID).evaluateAttributeExpressions(flowFile).getValue())
+                    .agentAddress(context.getProperty(V1TrapProperties.AGENT_ADDRESS).evaluateAttributeExpressions(flowFile).getValue())
+                    .genericTrapType(getGenericTrapType(context, flowFile))
+                    .specificTrapType(context.getProperty(V1TrapProperties.SPECIFIC_TRAP_TYPE).asInteger())
+                    .build();
+
+            pduFactory = new V1TrapPDUFactory(snmpResourceHandler.getTarget(), trapConfiguration, startTime);
```

Comment:
The factory should still be created in the init() and should be kept context-independent.
The instead, the trap configuration (the creation of which should be kept here as it is context-dependent) should be passed as a parameter to the trap-creating method.